lib/api/objectPutPart.js

```diff
@@ -205,6 +205,10 @@ function objectPutPart(authInfo, request, streamingV4Params, log,
                         return next(null, destinationBucket,
                             objectLocationConstraint, cipherBundle, splitter,
                             partInfo);
+                    } else if (partInfo && partInfo.dataStoreType === 'gcp') {
```

Comment:
trivial, but these two `else if`s could be combined with something like `['azure', 'gcp'].includes(partInfo.dataStoreType)`